plugins/home/src/api/config.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Visit, VisitsApiQueryParams } from './VisitsApi';
+import { Config } from '@backstage/config';
+
+/**
+ * Reads a single FilterBy config.
+ *
+ * @param config - The single config object
+ *
+ * @public
+ */
+export function readFilterByConfig(config: Config): {
+  field: keyof Visit;
+  operator: '<' | '<=' | '==' | '!=' | '>' | '>=' | 'contains';
+  value: string | number;
+} {
+  try {
+    const field = config.getString('field') as keyof Visit;
+    const operator = config.getString('operator') as
+      | '<'
+      | '<='
+      | '=='
+      | '!='
+      | '>'
+      | '>='
+      | 'contains';
+    const value = config.getString('value');
+    return { field, operator, value };
+  } catch (error) {
+    throw new Error(`Invalid config, ${error}`);
```

Comment:
I have a concern here.
I would expect that throwing this error here breaks the UI?

We want to capture the invalid config during the build and report that back to the user who is configuring the filters. 

And here instead of throwing an error, we probably just want to default to no filtering.

Reply:
I can change the catch implementation to use "no filtering". 
I'm not sure how to capture the invalid config during the build, is there an example?

Reply:
This will not break the UI because this code is running inside `readFilterByConfigs` and incase an error was thrown we will return undefined there 

`export function readFilterByConfigs(
  configs: Config[],
): VisitsApiQueryParams['filterBy'] | undefined {
  try {
    return configs.map(readFilterByConfig);
  }
  catch {
    return undefined;
  }
}`

then, in `context.tsx` we get the value and use it there.

> We want to capture the invalid config during the build and report that back to the user who is configuring the filters.

I'm not sure I get this one, currently, I saw an error in the log but the UI doesn't break

@tikabom - WDYT?

Reply:
@shmaram will let @tikabom answer your questions but just noting that we don't have tests for these helper functions `readFilterByConfig` and `readFilterByConfigs`. Can we add that as part of this PR? that could also test the case of invalid config by the way!

Reply:
@emmaindal - Added your suggested tests. 
Will wait for @tikabom to review as well.

BTW - does anyone know why the "E2E Linux" check failed?

Reply:
Hello @shmaram sorry for late replies here just coming back after some time off. Some test fails due to prettier errors in 

```
plugins/home/src/api/config.test.ts
plugins/home/src/api/config.ts
```

can you run prettier against these files? `yarn prettier --write plugins/home/src/api`

Reply:
@shmaram my bad! I didn't realise that `readFilterByConfig` was being called by `readFilterByConfigs` which is then called in `Content.tsx`! Could we rename these functions to `readFilterConfig` and `createFilterByQueryParamFromConfig`?

Another question here: Do we want to ignore all filter configs even if only a subset of them are bad? For example if there were 3 different filter configs and if only one of them were bad, throwing an error would mean that `filterBy` would always be `undefined`. What do you think @emmaindal ? Should we still be accepting the correctly configured filters?

Also it seems like bad app-configs aren't automatically captured as part of `yarn build` and need to be captured explicitly by running `@backstage/cli config:check`. @shmaram Could you add this to the README as a note? Thank you!

Reply:
I agree with you.. @shmaram let us know if you have any specific use cases in mind why it makes sense to ignore all filters.

Reply:
Hi @emmaindal @tikabom 
Thanks for the comments! 
The reason that I thought we should ignore all filters is to avoid misunderstanding for the developer, let's say they add 10 filters and 1 was bad, it would be very difficult to know that 9 are good but 1 didn't pass, but if no filters were set than it would be easy to see. 
Your suggestion works well also, I can go with both, so I will let you choose :)

Reply:
Hi @shmaram, let's apply filters that are set correctly. Since each filter config is processed separately anyway, you can throw the Error with the Config object currently being processed.

That + README update, and you're good to 🚢

Reply:
Done 😄